include/bamit/IntervalNode.hpp

```diff
@@ -382,15 +387,15 @@ inline std::streamoff get_overlap_records(seqan3::sam_file_input<traits_type, fi
         get_overlap_file_position(node_list[std::get<0>(start)], std::get<1>(start), std::get<1>(end), file_position);
     else // Searching across multiple chromosomes.
     {
-        for (uint32_t i = std::get<0>(start); i < std::get<0>(end); ++i)
+        for (uint32_t i = std::get<0>(start); i < (uint32_t) std::get<0>(end); ++i)
```

Comment:
Is this cast necessary? I figured since `end` is a tuple of two `uint32_t` types, it wouldn't be necessary to have.

Reply:
I just made this, because of the warning I got
```bash
/Users/lydia/Repos/BAMIntervalTree/src/../include/bamit/IntervalNode.hpp:385:49: warning: comparison of integer expressions of different signedness: 'uint32_t' {aka 'unsigned int'} and 'std::__tuple_element_t<0, std::tuple<int, int> >' {aka 'const int'} [-Wsign-compare]
  385 |         for (uint32_t i = std::get<0>(start); i < std::get<0>(end); ++i)
      |                                               ~~^~~~~~~~~~~~~~~~~~
```
See https://github.com/joshuak94/BAMIntervalTree/pull/38/commits/8a0bffa5b591aca8bc7fc6967bdd21cdcd44cbf2